src/main/java/com/plugin/HttpOutput.java

```diff
@@ -66,42 +122,51 @@ public void stop() {
 
 	@Override
 	public void write(List<Message> msgs) throws Exception {
-		for (Message msg : msgs) {
-			writeBuffer(msg.getFields());
-		}
+
+        for (List<Message> partition: ListUtils.partition(msgs, HTTP_BATCH_SIZE)) {
+            List<Map<String, Object>> payload = new FastArrayList();
+            for (Message msg : partition) {
+                payload.add(msg.getFields());
+            }
+            executeRequest(RequestBody.create(
+                    JSON,
+                    this.gson.toJson(payload)
+            ));
+        }
+
 	}
 
 	@Override
 	public void write(Message msg) throws Exception {
-		if (shutdown) {
-			return;
-		}
 
-		writeBuffer(msg.getFields());
+		executeRequest(
```

Comment:
To improve the readability, maybe use `this.executeRequest` instead of `executeRequest`
```suggestion
		this.executeRequest(
```